### DISCUSSION TITLE
Elixir GRPC instrumentation

### DISCUSSION THREAD

**Discussion by diegogl12** [open-telemetry/opentelemetry-erlang-contrib#170] [Ideas]
I'm currently using the [elixir grpc](https://github.com/elixir-grpc/grpc) lib and noticed that there is no support for this lib in the OTEL world.
Currently, here we can find the [grpcbox](https://github.com/open-telemetry/opentelemetry-erlang-contrib/tree/main/instrumentation/opentelemetry_grpcbox) support, but in my understanding, it's a different lib and approach from the [elixir grpc](https://github.com/elixir-grpc/grpc).

I was exploring the [GRPC.ServerInterceptor](https://github.com/elixir-grpc/grpc/blob/master/lib/grpc/server/interceptor.ex) and [GRPC.ClientInterceptor](https://github.com/elixir-grpc/grpc/blob/master/lib/grpc/client/interceptor.ex) that could be useful ways to handle this OTEL collection.

I was also inspired by the [Prometheus support](https://github.com/elixir-grpc/grpc-prometheus) for the [elixir gRPC](https://github.com/elixir-grpc/grpc).

I was energized to implement new instrumentation to have this support. Any thoughts?

**Comment by tsloughter** [comment]
A library for Elixir grpc would be great! And right, grpcbox is unrelated to elixir-grpc. It just happens to be the one I use (and OpenTelemetry exporter uses) so I wrote the instrumentation library for it.

**Reply by diegogl12** [reply]
Awesome... I'll start to look into it! 😊

**Comment by smashedtoatoms** [comment]
Would this deal with [this](https://community.fly.io/t/anyone-here-using-fly-postgres-with-opentelemetry-ecto/7198/2)?  I ran into this last year and have had no time to deal with it.

**Reply by diegogl12** [reply]
I don't think so. You had an issue related to ecto (if I understood). Here I'm proposing new instrumentation for the lib.

**Reply by tsloughter** [reply]
Maybe confusion around RPC/GRPC?

**Reply by smashedtoatoms** [reply]
Yep, The g never registered. Apologies.

**Comment by grzuy** [comment]
For what is worth, I think an `opentelemetry_grpc` was implemented in https://github.com/open-telemetry/opentelemetry-erlang-contrib/pull/554 recently.
I think not yet released/published to hex.pm, if I am not mistaken.
